src/s3/util/fileViewerManager.ts

```diff
@@ -59,14 +73,14 @@ export class S3FileViewerManager {
                         overwrite
                     )
                     if (response === cancelUpload) {
-                        download = false
+                        upload = false
                     }
                 }
 
-                if (download) {
-                    if (!(await activeTab.uploadChangesToS3())) {
+                if (upload) {
+                    if (!(await this.uploadChangesToS3(activeTab))) {
                         this.window.showErrorMessage(
-                            'Error uploading file to S3. Please try and resave this edit mode file'
+                            'Error uploading file to S3. Changes were not saved back to S3. Please try and resave this edit mode file'
```

Comment:
Just making sure: does `onDidSaveTextDocument` trigger if you save a file that isn't dirty?

Reply:
It does